src/models/transaction/TransferTransaction.ts

```diff
@@ -137,6 +152,40 @@ export class TransferTransaction extends Transaction {
     return this._mosaics!.map((_) => _.mosaicId);
   }
 
+  /**
+   * returns mosaic details without sorting through array
+   * @returns MosaicTransferable
+   */
+  public mosaicDetails = (): XEM | CACHE | MosaicTransferable => {
+    if (this.containsMosaics()) {
+      return mosaicDetails(this);
+    } else {
+      return xemDetails(this);
+    }
+  };
+
+  /**
+   * Create a CacheTransferTransaction object
+   * @param txType - TxType.xem, .cache, .other
+   * @param recipient
```

Comment:
finish adding comments. Also again i think we should be verbose whereever possible: txType maybe instead transactionType